README.md

```diff
@@ -10,6 +10,192 @@ pulling data from several locations:
 - whenever a user's homedir is created, it is populated with
   the contents of the (signed) [dotfiles] repository.
 
+## Building ##
+
+Build all image types:
+
+```
+make all
+```
+
+Or build a specific image type such as vagrant:
+```
+make vagrant
+```
+
+All artifacts will be placed in ```$PWD/dist```.
+
+## Releasing ##
+
+1. Copy config sample and populate with your credentials as desired:
+
+    ```
+    cp config.sample.json config.json
+    vim config.json
+    ```
+
+2. Build, sign, and publish all image types
+    ```
+    make build release
+    ```
+## Development ##
+
+Generally you want to use our pre-built images.
+
+The following will get you a standalone shell-server locally to develop on.
+
+You will need a #! account to use these, as they authenticate users against
+our NSS services by default.
+
+These are intended to mirror the flow of you setting up a federated server as
+per the [Deployment] section of this Readme.
+
+For a -fully- local development infrastructure consider the e2e
+testing/development suite found in the [hashbang] repo.
+
+### Docker ###
+
+Start server:
+```
+docker run \
+  -ti \
```

Comment:
meganit: `docker run -it` (see :74 for `docker exec -it`) sounds better in my head :zipper_mouth_face: